### PR TITLE
Update to ICSharpCode.Decompiler 7.1.0.6543

### PR DESCRIPTION
**Pull request by christophwille**
For new language features (mostly C# 9.0) see https://github.com/icsharpcode/ILSpy/releases/tag/v7.0

IAssemblyResolver implemented as per https://github.com/icsharpcode/ILSpy/blame/dfd82de1afdb7d897d1c0084856a84197ac8fd20/ICSharpCode.Decompiler/Metadata/UniversalAssemblyResolver.cs#L213-L221

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=53304) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=53304) before we can accept your contribution.<br/><br/>:x: christophwille [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=53304)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn?pullRequest=53304) it.</sub>

File: src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs [hunk 1]
```diff
@@ -43,10 +44,14 @@ void BuildReferenceCache()
             }
         }
 
-        bool IAssemblyResolver.IsGacAssembly(IAssemblyReference reference)
+        public Task<PEFile> ResolveAsync(IAssemblyReference name)
         {
-            // This method is not called by the decompiler
-            throw new NotSupportedException();
+            return Task.Run(() => Resolve(name));
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs:49]: Why the Task.Run here? This may deserve a comment.

**Comment by jasonmalinowski** [comment]
Should we have this target 16.11? @sharwell?

File: src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (christophwille) [reply src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs:49]: As pointed out, I took it from our implementation - any other way is fine with me that fits your needs.

**Comment by christophwille** [comment]
We release https://github.com/icsharpcode/ILSpy/releases/tag/v7.1 yesterday. To update the PR, it would need to be added to the feed.

**Comment by chuckries** [comment]
@christophwille @sharwell 
Is this PR being held up by anything? VS Debugger depends on the ICSharpCode.Decompiler inserted by Roslyn for debug time decompilation (with decompiled debug info). The version currently shipping with VS 2022 Preview (6.1.0) can NRE during debug info generation which causes full decompilation to fail for some assemblies. The issue appears to be fixed in both 7.0 and 7.1 version of the Decompiler.

TL;DR: VS Debugger needs this PR or a similar update PR to get merged into Roslyn and inserted into VS.

**Comment by sharwell** [comment]
> To update the PR, it would need to be added to the feed.

I submitted a request to add them.

**Comment by sharwell** [comment]
@christophwille I'm fixing the tests and will push the updates to this PR.

**Comment by christophwille** [comment]
This has now been sitting around for quite some time even after Sam's update to the latest infra changes. Is there a plan to merge this?

**Comment by sharwell** [comment]
I'll ping the internal thread for this

**Comment by JoeRobich** [comment]
Test Insertion - https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/352300 (microsoft)

**Comment by JoeRobich** [comment]
@sharwell Test insertion was passing.

**Comment by christophwille** [comment]
Should we close this pull request as there is no progress (indication) for months?

**Comment by sharwell** [comment]
Finally had a meeting today. Should have this resolved/merged by the end of the month at the latest.

**Review by sharwell — APPROVED** [review]
